File: src/js/core/support/custom_types/listNode.js
class SingleLinkedNode{
    /**
     * Constructor for the Single Directional Linked Node
     * @param value {null || Number}
     * @param next {SingleLinkedNode || null}
     */
    constructor(value = null, next= null) {
        this.val = value;
        this.next = next;
    }

    /**
     * Set the value of the node
     * @param value {Number || String}
     */
    set setVal(value){
        this.val = value;
    }

    /**
     * Return the value of the node
     * @return {null|Number}
     */
    get getVal(){
        return this.val;
    }

    /**
     * Set the next for SingleLinkedNode
     * @param next {SingleLinkedNode}
     */
    set setNext(next){
        this.next = next;
    }

    /**
     * Return the next node
     * @return {SingleLinkedNode|null}
     */
    get getNext(){
        return this.next
    }
}


module.exports = {
    default: SingleLinkedNode
}

File: src/java/core/support/custom_types/ListNode.java
package support.custom_types;

/**
 * Defines a single directional ListNode
 */
public class ListNode {
    /**
     * Integer value of the node
     */
    public int val;
    /**
     * The next node in the list. If there is no node afterwards, ListNode.next
     * is set to null
     */
    public ListNode next;

    /**
     * Class variables left as undefined if not initiated
     */
    public ListNode(){
    }

    /**
     * Defines the value at the current node.
     * Set the next property to null
     * @param val Value at the current node
     */
    public ListNode(int val){
        this(val, null);
    }

    /**
     *
     * @param val Value at the current node
     * @param next Pointer to next node
     */
    public ListNode(int val, ListNode next){
        this.val = val;
        this.next = next;
    }
}
